hack/lib/verify-generated.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Short-circuit if verify-generated.sh has already been sourced.
+[[ $(type -t kube::verify::generated::loaded) == function ]] && return 0
+
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# This function verifies whether generated files are up-to-date. The first two
+# parameters are messages that get printed to stderr when changes are found,
+# the rest are the function or command and its parameters for generating files
+# in the work tree.
+#
+# Example: kube::verify::generated "Mock files are out of date" "Please run 'hack/update-mocks.sh'" hack/update-mocks.sh
+kube::verify::generated() (
```

Comment:
This is tricky syntax.  It surprised even me.  Can I arm-twist you into something more obvious:

```
kube::verify::generated() {
  ( # a subshell prevents environment changes from leaking out of this function
       local failure_header=$1
       shift

       # etc.
  )
}
```

Reply:
Sure, if that's preferred. Defining commands as a single sub-shell is how I learned it, but I guess it's not that common in Kubernetes.

Updated.